packages/auth/src/access/service.test.ts

```diff
@@ -0,0 +1,123 @@
+import crypto from 'crypto'
+import nock from 'nock'
+import Knex, { Transaction } from 'knex'
+import { v4 } from 'uuid'
+import { createTestApp, TestContainer } from '../tests/app'
+import { truncateTables } from '../tests/tableManager'
+import { Config } from '../config/app'
+import { IocContract } from '@adonisjs/fold'
+import { initIocContainer } from '../'
+import { AppServices } from '../app'
+import { AccessService } from './service'
+import { Grant, GrantState, StartMethod, FinishMethod } from '../grant/model'
+import { Action, AccessType, AccessRequest } from './types'
+
+describe('Access Service', (): void => {
+  let deps: IocContract<AppServices>
+  let appContainer: TestContainer
+  let accessService: AccessService
+  let knex: Knex
+  let trx: Transaction
+
+  beforeAll(
+    async (): Promise<void> => {
+      deps = await initIocContainer(Config)
+      accessService = await deps.use('accessService')
+      knex = await deps.use('knex')
+      appContainer = await createTestApp(deps)
+    }
+  )
+
+  afterEach(
+    async (): Promise<void> => {
+      await truncateTables(knex)
+    }
+  )
+
+  afterAll(
+    async (): Promise<void> => {
+      nock.restore()
+      await appContainer.shutdown()
+    }
+  )
+
+  const BASE_GRANT = {
+    state: GrantState.Pending,
+    startMethod: [StartMethod.Redirect],
+    continueToken: crypto.randomBytes(8).toString('hex').toUpperCase(),
+    continueId: v4(),
```

Comment:
This is used to create a `Grant` since one needs to exist in order to have a valid foreign key for the `grantId` in `Access` when it is subsequently created.